src/debugger/nodeDebugWrapper.ts

```diff
@@ -1,194 +1,246 @@
+        this.mobilePlatformOptions = {
+            projectRoot: this.projectRootPath,
+            platform: args.platform,
         };
 
-        // Intecept the "launchRequest" instance method of NodeDebugSession to interpret arguments
-        const originalNodeDebugSessionDisconnectRequest = nodeDebug.NodeDebugSession.prototype.disconnectRequest;
-        function customDisconnectRequest(response: any, args: any): void {
-            try {
-                // First we tell the extension to stop monitoring the logcat, and then we disconnect the debugging session
-                const remoteExtension = RemoteExtension.atProjectRootPath(projectRootPath);
-                remoteExtension.stopMonitoringLogcat()
-                    .finally(() => originalNodeDebugSessionDisconnectRequest.call(this, response, args))
-                    .done(() => { }, reason => // We just print a warning if something fails
-                        process.stderr.write(`WARNING: Couldn't stop monitoring logcat: ${reason.message || reason}\n`));
-            } catch (exception) {
-                // This is a "nice to have" feature, so we just fire the message and forget. We don't event handle
-                // errors in the response promise
-                process.stderr.write(`WARNING: Couldn't stop monitoring logcat. Sync exception: ${exception.message || exception}\n`);
-                originalNodeDebugSessionDisconnectRequest.call(this, response, args);
+        // Start to send telemetry
+        this.telemetryReporter.reassignTo(new ExtensionTelemetryReporter(
+            this.appName, this.version, Telemetry.APPINSIGHTS_INSTRUMENTATIONKEY, this.projectRootPath));
+
+        // Create a server waiting for messages to re-initialize the debug session;
+        const debugServerListeningPort = this.createReinitializeServer(debugSession, args.internalDebuggerPort);
+        args.args = [debugServerListeningPort.toString()];
+
+        Log.SetGlobalLogger(new NodeDebugAdapterLogger(this.vscodeDebugAdapterPackage, debugSession));
+    }
+
+    /**
+     * Runs logic needed to attach.
+     * Attach should:
+     * - Enable js debugging
+     */
+    private attachRequest(nodeAdapterRequest: (request: any, args: any) => void, debugSession: NodeDebugSession, request: any, args: any, mobilePlatform: any): Q.Promise<void> {
+        return TelemetryHelper.generate("attach", (generator) => {
+            return Q({})
+                .then(() => {
+                    generator.step("mobilePlatform.enableJSDebuggingMode");
+                    if (mobilePlatform) {
+                        return mobilePlatform.enableJSDebuggingMode();
+                    } else {
+                        Log.logMessage("Debugger ready. Enable remote debugging in app.");
+                    }
+                }).then(() =>
+                    nodeAdapterRequest.call(debugSession, request, args))

```

Comment:
Do you know if the node debug adapter returns a promise? Might be worth returning the value here anyway.